### PR TITLE
feat(design-tokens): Add support for new color slots - calculation, dimension & measure

### PR DESCRIPTION
**Pull request by lukelooker**
Adds three new theme "slots" for so-called stateful colors. These colors will be used for `IconButton` and `ListItem` colors in an upcoming PR (#2165).

## Developer Checklist [ℹ️](https://github.com/looker-open-source/components/blob/main//CONTRIBUTING.md#developer-checklist)

- [x] 🖼 Image Snapshot coverage
- [ ] 📚 Documentation updated
- [x] 🧳 Bundle size impact addressed
- [x] 🏁 Performance impacts addressed

### CODE REVIEW

File: packages/design-tokens/src/color/defaults.ts [hunk 1]
```diff
@@ -36,16 +36,23 @@ import {
 } from '../legacy/palette'
 import { CoreColors, IntentColors } from './types'
 
+const calculation = '#319220'
+const dimension = '#31689E'
+const measure = '#C2772E'
```
[the remark this answers is not in the document]
Comment (ElliotTheHuman) [reply packages/design-tokens/src/color/defaults.ts:41]: I like these as a start at least. The only question I'd have would be whether or not to use "calc" or "calculation", but I think the explicitness of "calculation" is worth the extra characters.

File: packages/design-tokens/src/color/types/stateful.ts [hunk 2]
```diff
@@ -36,8 +42,14 @@ export interface StatefulColorChoices {
   border: string
 }
 
+export interface ExtendedStateColors
```
Comment (ElliotTheHuman) [inline packages/design-tokens/src/color/types/stateful.ts:45]: nit: This name seems super close to `ExtendedStatefulColor`. Would changing this to something like `FieldTypeColors` work?

(Feel free to disregard this comment)

File: packages/design-tokens/src/color/defaults.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (lukelooker) [reply packages/design-tokens/src/color/defaults.ts:41]: Yeah - for names like this, while perhaps annoyingly long I think explicit wins :)

**Review by ElliotTheHuman — DISMISSED** [review]
Left a minor nit about the `ExtendedStatefulColor` type and `ExtendedStateColors` interface, but other than that, lgtm!

**Review by ElliotTheHuman — APPROVED** [review]
:shipit:

**Pull request merged**